extensions/ql-vscode/src/remote-queries/shared/variant-analysis.ts

```diff
@@ -1,5 +1,92 @@
+import { Repository } from './repository';
+
+export interface VariantAnalysisSubmission {
+  startTime: number,
+  controllerRepoId: number,
+  actionRepoRef: string,
+  query: {
+    name: string,
+    filePath: string,
+    language: VariantAnalysisQueryLanguage,
+
+    // Base64 encoded query pack.
+    pack: string,
+  },
+  databases: {
+    repositories?: string[],
+    repositoryLists?: string[],
+    repositoryOwners?: string[],
+  }
+}
+
+export interface VariantAnalysis {
+  id: number,
+  controllerRepoId: number,
+  query: {
+    name: string,
+    filePath: string,
+    language: VariantAnalysisQueryLanguage
+  },
+  databases: {
+    repositories?: string[],
+    repositoryLists?: string[],
+    repositoryOwners?: string[],
+  },
+  status: VariantAnalysisStatus,
+  actionsWorkflowRunId?: number,
+  failureReason?: VariantAnalysisFailureReason,
+  scannedRepos?: VariantAnalysisScannedRepository[],
+  skippedRepos?: VariantAnalysisSkippedRepositories
+}
+
+export enum VariantAnalysisQueryLanguage {
+  CSharp = 'csharp',
+  Cpp = 'cpp',
+  Javascript = 'javascript',
+  Go = 'go',
+  Python = 'python',
+  Ruby = 'ruby'
+}
+
 export enum VariantAnalysisStatus {
   InProgress = 'inProgress',
   Succeeded = 'succeeded',
   Failed = 'failed',
 }
+
+export enum VariantAnalysisFailureReason {
+  NoReposQueried = 'noReposQueried',
+  InternalError = 'internalError',
+}
+
+export enum VariantAnalysisRepoStatus {
+  Pending = 'pending',
+  InProgress = 'inProgress',
+  Succeeded = 'succeeded',
+  Failed = 'failed',
+  Canceled = 'canceled',
+  TimedOut = 'timedOut',
+}
+
+export interface VariantAnalysisScannedRepository {
+  repository: Repository,
+  analysisStatus: VariantAnalysisRepoStatus,
+  resultCount?: number,
+  artifactSizeInBytes?: number,
+  failureMessage?: string
+}
+
+export interface VariantAnalysisSkippedRepositories {
+  accessMismatchRepos?: VariantAnalysisSkippedRepositoryGroup,
+  notFoundRepos?: VariantAnalysisSkippedRepositoryGroup,
+  noCodeql_dbRepos?: VariantAnalysisSkippedRepositoryGroup,
+  overLimitRepos?: VariantAnalysisSkippedRepositoryGroup
+}
+
+export interface VariantAnalysisSkippedRepositoryGroup {
+  repoCount: number,
+  repositories: Array<{
+    id: number,
```

Comment:
We won't have an ID for repos which are not found, so this might need to be optional (or there needs to be a separate type for the `notFoundRepos`)

Reply:
I've gone with making it optional for now - I think the view will only use the nwo anyway.